src/stickybits.js

```diff
@@ -61,58 +61,56 @@ Stickybit.prototype.setStickyPosition = function setStickyPosition() {
   => based on window scroll
 */
 Stickybit.prototype.manageStickiness = function manageStickiness() {
-  const parent = this.el.parentNode;
-  const pv = this.positionStickyVal;
+  // cache variables
+  const el = this.el;
+  const parent = el.parentNode;
+  const pv = this.positionVal;
   const vp = this.vp;
   const offset = this.offset;
   const styles = this.styles;
-  const classes = this.el.classList;
+  const classes = el.classList;
   const win = window;
+
+  // setup css classes
   parent.classList.add('js-stickybit-parent');
-  const stickyBitStart = this.el.getBoundingClientRect().top;
-  const stickyBitStop = (stickyBitStart + parent.offsetHeight) -
-    (this.el.offsetHeight - offset);
   const stickyClass = 'js-is-sticky';
   const stuckClass = 'js-is-stuck';
 
-  // manage stickiness
-  function stickiness() {
+  // manageState
+  const stickyStart = parent.getBoundingClientRect().top;
+  const stickyStop = (stickyStart + parent.offsetHeight) -
+      (el.offsetHeight - offset);
+  let state = 'default';
+
+  this.manageState = () => {
     const scroll = win.scrollY || win.pageYOffset;
-    const hasStickyClass = classes.contains(stickyClass);
-    const hasStuckClass = classes.contains(stuckClass);
-    if (scroll < stickyBitStart) {
-      if (hasStickyClass) {
+    if ((scroll > stickyStart) && (scroll < stickyStop) && state === 'default') state = 'sticky';
```

Comment:
i would keep this logic and the raf logic together so that there's not two sets of if statements

```js
if ((scroll < stickyStart && state === 'sticky')) {
  this.state = 'default'
  requestAnimationFrame(() => {
    // do stuff
  })
  return
}